src/main/java/seedu/duke/Parser.java

```diff
@@ -44,19 +59,49 @@ public static Command checkAction(String inputCommand) {
             break;
         case "select":
             if (isProjectListView) {
-                projectIndex = Integer.parseInt(inputCommand.split(" ")[1]) - 1;
+                projectIndex = Integer.parseInt(inputs[1]) - 1;
                 commandType = new ProjectSelectCommand(projectIndex);
-            } 
+            } else {
+                taskIndex = Integer.parseInt(inputs[1]) - 1;
+                commandType = new TaskSelectCommand(taskIndex, projectIndex);
+            }
             break;
         case "project":
-            String deadline = inputCommand.split(" /by ")[1];
-            projectDescription = inputCommand.split(" /by ", 2)[0].split(" ", 2)[1];
-            commandType = new ProjectCommand(projectDescription, deadline);
+            if (isProjectListView) {
+                for (int i = 1; i < inputs.length; i++) {
+                    projectDescription += inputs[i];
+                }
+                commandType = new ProjectCommand(projectDescription);
+                break;
+            } else {
+                System.out.println("Not in Project View!");
+            }
             break;
         case "task":
-            deadline = inputCommand.split(" /by ")[1];
-            projectDescription = inputCommand.split(" /by ", 2)[0].split(" ", 2)[1];
-            commandType = new TaskCommand(projectDescription, deadline, projectIndex);
+            if (isProjectListView) {
+                System.out.println("Not in Task View!");
+            } else {
+                for (int i = 1; i < inputs.length; i++) {
+                    projectDescription += inputs[i];
+                }
+                commandType = new TaskCommand(projectDescription, projectIndex);
+            }
+            break;
+        case "delete":
+            if (isProjectListView) {
+                //Implement delete project here - Small Sam
```

Comment:
@samuelchristopher consider what happens to projectIndex pointer if project gets deleted.